### PR TITLE
Issue/2939 purge old devices (Connect #2939)

### PR DESCRIPTION
**Pull request by stellanl**
## Checklist
* [x] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 1]
```diff
@@ -68,21 +74,56 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
...
+        log.info("Found " + deviceList.size() + " old Devices");
+
+        for (Device d: deviceList) { //Clean up everything referencing this device
+        	
+        	List<DeviceSurveyJobQueue> djql = dsjqDao.get(d.getPhoneNumber(), d.getEsn(), d.getAndroidId());
+        	if (djql.size() > 0) {
+        		log.fine("Deleting " + djql.size() + " form assignments for device " + d.getKey().getId());
+        		dsjqDao.delete(djql);
+        	}
+        	List<DeviceFileJobQueue> dfql = dfjqDao.listByDeviceId(d.getKey().getId());
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: Maybe you should extract d.getKey().getId() to var

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 2]
```diff
@@ -31,18 +31,24 @@
...
 import com.gallatinsystems.notification.helper.NotificationHelper;
 import com.gallatinsystems.survey.dao.DeviceSurveyJobQueueDAO;
+import com.gallatinsystems.survey.dao.SurveyAssignmentDAO;
 import com.gallatinsystems.survey.dao.SurveyDAO;
 import com.gallatinsystems.survey.dao.SurveyTaskUtil;
 import com.google.appengine.api.datastore.Key;
 
 
+/**
+ * @author stellan
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: is this comment necessary? Most codebases don't add the name to the files as git tracks all the changes made to the file. It's just a question of habit, I guess.

File: GAE/src/com/gallatinsystems/survey/dao/SurveyAssignmentDAO.java [hunk 3]
```diff
@@ -16,13 +16,52 @@
...
         super(SurveyAssignment.class);
     }
+    
+    /**
+     * Return a list of survey assignments containing a specified device
+     *
+     * @return list of assignments
+     */
+    public List<SurveyAssignment> listAllContainingDevice(Long devId) {
+    	List<SurveyAssignment> all = list(Constants.ALL_RESULTS);
```
Comment (valllllll2000) [inline GAE/src/com/gallatinsystems/survey/dao/SurveyAssignmentDAO.java]: I think it's possible to get all assignments for a deviceId, including a screenshot from the console
screenshot from 2019-01-24 18-10-22

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java (hunk 1 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: We do it this way in lots of places. It is not expensive...

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java (hunk 2 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: Ah, didn't notice Eclipse added that

File: GAE/src/com/gallatinsystems/survey/dao/SurveyAssignmentDAO.java (hunk 3 above)
Comment (stellanl) [reply GAE/src/com/gallatinsystems/survey/dao/SurveyAssignmentDAO.java]: Interesting, so the index generation looks inside lists. I never knew. Optimised that.

**Pull request merged**